deps/v8/src/debug/debug-coverage.cc

```diff
@@ -584,30 +610,40 @@ std::unique_ptr<Coverage> Coverage::Collect(
     result->emplace_back(script_handle);
     std::vector<CoverageFunction>* functions = &result->back().functions;
 
-    std::vector<SharedFunctionInfo> sorted;
+    std::vector<SharedFunctionInfoAndCount> sorted;
 
     {
       // Sort functions by start position, from outer to inner functions.
       SharedFunctionInfo::ScriptIterator infos(isolate, *script_handle);
       for (SharedFunctionInfo info = infos.Next(); !info.is_null();
            info = infos.Next()) {
-        sorted.push_back(info);
+        sorted.emplace_back(info, counter_map.Get(info));
       }
-      std::sort(sorted.begin(), sorted.end(), CompareSharedFunctionInfo);
+      std::sort(sorted.begin(), sorted.end());
```

Comment:
Is perhaps part of the problem that this should be `std::stable_sort(sorted.begin(), sorted.end());`?

Reply:
I think there's a chance that the sorting behavior wouldn't flip/flop if we used `stable_sort`. But I'm not sure that we know if the "counted" function element is always added first, I like that we're more explicit and bring functions with coverage to the top of the sorting order.

Reply:
Unfortunately stable_sort wouldn't help. The order of shared function infos on the heap is nondeterministic and likewise the initial (unsorted) order in `sorted`. A stable sort would just preserve that initial order for "same" elements.

The fix in this change is to also consider the SFI's call count while sorting. As described in the comment, it's somewhat of a hack, but works. The cleaner solution, potentially in the future, would be to remove the optimization to omit "irrelevant" coverage.